### PR TITLE
Bugsnag.MetaData, Dotenv, and removing AppVersion

### PR DESCRIPTION
**Pull request by pbrisbin**
This PR makes three interrelated changes that together simplify how we handle metadata
(particularly version) in Bugsnag reporting. Ultimately, this gets our snags sharing the exact
same metadata as our metrics have tags and our logging has context. :handshake: 

Please see individual commit messages for more details.

### CODE REVIEW

File: library/Freckle/App/Bugsnag/MetaData.hs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+collectMetaDataFromThreadContext =
+  liftIO $ metaData "context" . Aeson.toList <$> myThreadContext
+
+-- | Merge 'MetaData' into the 'Event'
+--
+-- The given metadata will be combined with what already exists using '(<>)',
+-- preserving the incoming values on collisions.
+--
+mergeMetaData :: MetaData -> BeforeNotify
+mergeMetaData md = updateEvent $ metaDataL <>~ md
```
Comment (pbrisbin) [inline library/Freckle/App/Bugsnag/MetaData.hs:90]: This little guy right here. This alone makes learning Lens worthwhile.

File: library/Freckle/App/Dotenv.hs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+-- | Call 'loadFrom' with @.env@
+load :: HasCallStack => IO ()
+load = loadFrom ".env"
+
+-- | Call 'loadFrom' with @.env.test@
+loadTest :: HasCallStack => IO ()
+loadTest = loadFrom ".env.test"
+
+-- | Load the given file, asserting it contains all variables in @.env.example@
```
Comment (pbrisbin) [inline library/Freckle/App/Dotenv.hs:25]: Should add "...when present".

I decided to support a lack of `.env.example` (for now), just to ease transition of existing apps.

File: tests/Freckle/App/Bugsnag/MetaDataSpec.hs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+          ]
+        expected = metaData
+          "test"
+          [ "foo" .= ("bar2" :: Text)
+          , "baz" .= object ["bat" .= ("quix2" :: Text)]
+          , "keep" .= ("me" :: Text)
+          , "add" .= ("me" :: Text)
+          ]
+
+
```
Comment (pbrisbin) [inline tests/Freckle/App/Bugsnag/MetaDataSpec.hs:65]:
```suggestion
```

File: library/Freckle/App/Bugsnag/MetaData.hs (hunk 1 above)
Comment (mjgpy3) [reply library/Freckle/App/Bugsnag/MetaData.hs:90]: Yes, `<>~` is awesome

File: library/Freckle/App/Dotenv.hs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+loadFrom :: HasCallStack => FilePath -> IO ()
+loadFrom name = do
+  path <- locateInParents name
+
+  let examplePath = takeDirectory path </> ".env.example"
+  exampleExists <- doesFileExist examplePath
+
+  void $ Dotenv.loadFile $ Dotenv.Config
+    { Dotenv.configPath = [path]
+    , Dotenv.configExamplePath = [ examplePath | exampleExists ]
```
Comment (mjgpy3) [inline library/Freckle/App/Dotenv.hs:43]: I like comprehensions for this.

**Review by mjgpy3 — APPROVED** [review]
LGTM!

**Pull request merged**